xdsl/dialects/llvm.py

```diff
@@ -71,6 +76,37 @@
 from xdsl.utils.hints import isa
 from xdsl.utils.str_enum import StrEnum
 
+LLVMCompatibleSignlessIntegerConstraint = AnyOf(
+    [
+        SignlessIntegerConstraint,
+        VectorOf(SignlessIntegerConstraint),
+    ]
+)
```

Comment:
But TBH at this point I think it's worth just inlining in the definition, similarly below
```suggestion
LLVMCompatibleSignlessIntegerConstraint =(
        SignlessIntegerConstraint |
        VectorOf(SignlessIntegerConstraint)
)
```